x-pack/plugins/infra/public/alerting/log_threshold/rule_data_formatters.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import {
+  ALERT_EVALUATION_THRESHOLD,
+  ALERT_EVALUATION_VALUE,
+  ALERT_ID,
+  ALERT_START,
+} from '@kbn/rule-data-utils';
+import { modifyUrl } from '@kbn/std';
+import { fold } from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/lib/function';
+import { ObservabilityRuleTypeFormatter } from '../../../../observability/public';
+import {
+  ComparatorToi18nMap,
+  logThresholdRuleDataRT,
+  logThresholdRuleDataSerializedParamsKey,
+  ratioAlertParamsRT,
+} from '../../../common/alerting/logs/log_threshold';
+
+export const formatReason: ObservabilityRuleTypeFormatter = ({ fields }) => {
+  const reason = pipe(
+    logThresholdRuleDataRT.decode(fields),
+    fold(
+      () =>
+        i18n.translate('xpack.infra.logs.alerting.threshold.unknownReasonDescription', {
+          defaultMessage: 'unknown reason',
+        }),
+      (logThresholdRuleData) => {
+        const params = logThresholdRuleData[logThresholdRuleDataSerializedParamsKey][0];
+
+        const actualCount = fields[ALERT_EVALUATION_VALUE];
+        const groupName = fields[ALERT_ID];
+        const isGrouped = (params.groupBy?.length ?? 0) > 0;
+        const isRatio = ratioAlertParamsRT.is(params);
+        const thresholdCount = fields[ALERT_EVALUATION_THRESHOLD];
+        const translatedComparator = ComparatorToi18nMap[params.count.comparator];
+
+        if (isRatio) {
+          return i18n.translate('xpack.infra.logs.alerting.threshold.ratioAlertReasonDescription', {
+            defaultMessage:
+              '{isGrouped, select, true{{groupName}: } false{}}The ratio of log entries matching the configured conditions is {actualCount} ({translatedComparator} {thresholdCount}).',
```

Comment:
I'd be happy to hear ideas to improve these messages. I just went with relatively simple ones since I don't know how much space the new `<EuiDataGrid>`-based alerts table offers and where else the reason will be rendered.

Reply:
A slightly shorter message could be:
 `'{isGrouped, select, true{{groupName}: } false{}}Log entries ratio matching the configured conditions is {actualCount} ({translatedComparator} {thresholdCount}).',` 
or  
`'{isGrouped, select, true{{groupName}: } false{}}Log entries ratio is {actualCount} ({translatedComparator} {thresholdCount}).',`. Maybe too simple?

Reply:
I like the short version since we expect this to be rendered in a table along other context info. :+1: